contracts/BinaryOption.sol

```diff
@@ -19,125 +18,50 @@ contract BinaryOption is IERC20, IBinaryOption {
-    }
-
-    /* ========== VIEWS ========== */
-
-    function _claimableBalanceOf(
-        uint _bid,
-        uint price,
-        uint exercisableDeposits
-    ) internal view returns (uint) {
-        uint owed = _bid.divideDecimal(price);
-        uint supply = _totalClaimableSupply(exercisableDeposits);
-
-        /* The last claimant might be owed slightly more or less than the actual remaining deposit
-           based on rounding errors with the price.
-           Therefore if the user's bid is the entire rest of the pot, just give them everything that's left.
-           If there is no supply, then this option lost, and we'll return 0.
-           */
-        if ((_bid == totalBids && _bid != 0) || supply == 0) {
-            return supply;
-        }
-
-        /* Note that option supply on the losing side and deposits can become decoupled,
-           but losing options are not claimable, therefore we only need to worry about
-           the situation where supply < owed on the winning side.
-
-           If somehow a user who is not the last bidder is owed more than what's available,
-           subsequent bidders will be disadvantaged. Given that the minimum bid is 10^16 wei,
-           this should never occur in reality. */
-        require(owed <= supply, "supply < claimable");
-        return owed;
-    }
-
-    function claimableBalanceOf(address account) external view returns (uint) {
-        (uint price, uint exercisableDeposits) = market.senderPriceAndExercisableDeposits();
-        return _claimableBalanceOf(bidOf[account], price, exercisableDeposits);
-    }
-
-    function _totalClaimableSupply(uint exercisableDeposits) internal view returns (uint) {
-        uint _totalSupply = totalSupply;
-        // We'll avoid throwing an exception here to avoid breaking any dapps, but this case
-        // should never occur given the minimum bid size.
-        if (exercisableDeposits <= _totalSupply) {
-            return 0;
-        }
-        return exercisableDeposits.sub(_totalSupply);
-    }
+    bool public initialized = false;
 
-    function totalClaimableSupply() external view returns (uint) {
-        (, uint exercisableDeposits) = market.senderPriceAndExercisableDeposits();
-        return _totalClaimableSupply(exercisableDeposits);
+    function initialize(
+        string memory _name,
+        string memory _symbol
+    ) public {
+        require(!initialized, "Binary Option Market already initialized");
+        initialized = true;
+        name = _name;
+        symbol = _symbol;
+        market = BinaryOptionMarket(msg.sender);
     }
 
     /* ========== MUTATIVE FUNCTIONS ========== */
 
-    function _requireMinimumBid(uint bid) internal pure returns (uint) {
-        require(bid >= _MINIMUM_BID || bid == 0, "Balance < $0.01");
-        return bid;
+    function _requireMinimumAmount(uint amount) internal pure returns (uint) {
+        require(amount >= _MINIMUM_AMOUNT || amount == 0, "Balance < $0.01");
```

Comment:
why can you mint 0?

Reply:
I will check, but as you can see the original version had bid 0 in it too.

Reply:
I see there is even a unit tests in the original version that covered this and made sure bid zero is idempotent.

Reply:
The refund function also used to use this function. It allowed a user to refund their entire bid, but not refund an amount that would leave their bid at less than the required minimum.